sbpack/noncwl/Readme.md

```diff
@@ -53,15 +53,15 @@ to the Platform.
 ```
 $ sbpack_nf -h
 
-usage: sbpack_nf [-h] [--profile PROFILE] --appid APPID --entrypoint ENTRYPOINT --workflow-path WORKFLOW_PATH [--sb-package-id SB_PACKAGE_ID] [--sb-inputs SB_INPUTS] [--sb-outputs SB_OUTPUTS]
+usage: sbpack_nf [-h] [--profile PROFILE] --appid APPID --workflow-path WORKFLOW_PATH [--entrypoint ENTRYPOINT] [--sb-package-id SB_PACKAGE_ID] [--sb-inputs SB_INPUTS] [--sb-outputs SB_OUTPUTS]
                  [--sb-doc SB_DOC] [--dump-sb-app] [--no-package] [--json] [--sb-schema SB_SCHEMA]
 
 optional arguments:
   -h, --help            show this help message and exit
   --profile PROFILE     SB platform profile as set in the SB API credentials file.
   --appid APPID         Takes the form {user or division}/{project}/{app_id}.
   --entrypoint ENTRYPOINT
-                        Relative path to the workflow from the main workflow directory
+                        Relative path to the workflow from the main workflow directory. If not provided, 'main.nf' will be used if available. If not available, any '.nf' located in the workflow-path will be used.
```

Comment:
"If not available, the _first_ *.nf file located..."

Also, consider producing an error if inferring the correct entrypoint implicitly is not obvious (ie. there are multiple *.nf files and none of them is main.nf). This would force the user to explicitly choose one.

Reply:
I agree. I will implement changes to how implicit nextflow script search works.
1. If the "main.nf" script is present, use "main.nf"
2. If there is a single "*.nf" script in the root of the folder, use that file
3. If there are multiple "*nf" scripts in the root of the folder (and none is named "main.nf", raise an error